File: src/main/java/top/zbeboy/isy/web/util/SmallPropsUtils.java
package top.zbeboy.isy.web.util;

import org.apache.commons.lang3.math.NumberUtils;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by lenovo on 2016/9/22.
 */
public class SmallPropsUtils {

    /**
     * 参数ids ',' 分隔转list<Integer>
     * @param ids
     * @return list<Integer>
     */
    public static List<Integer> StringIdsToList(String ids){
        String[] idArr = ids.split(",");
        List<Integer> newIds = new ArrayList<>();
        for(String id:idArr){
            newIds.add(NumberUtils.toInt(id));
        }
        return newIds;
    }

    /**
     * 参数ids ',' 分隔转list<String>
     * @param ids
     * @return list<String>
     */
    public static List<String> StringIdsToStringList(String ids){
        String[] idArr = ids.split(",");
        List<String> newIds = new ArrayList<>();
        for(String id:idArr){
            newIds.add(id);
        }
        return newIds;
    }

    /**
     * 参数ids ',' 是否为Integer类型
     * @param ids
     * @return true or false
     */
    public static boolean StringIdsIsNumber(String ids){
        String[] idArr = ids.split(",");
        boolean isNumber = true;
        for(String id:idArr){
            if(!NumberUtils.isNumber(id)){
                isNumber = false;
                break;
            }
        }
        return isNumber;
    }
}

File: src/main/java/top/zbeboy/isy/web/util/SmallPropsUtils.kt
package top.zbeboy.isy.web.util

import org.apache.commons.lang3.math.NumberUtils
import java.util.*

/**
 * Created by zbeboy 2017-11-26 .
 **/
open class SmallPropsUtils {
    companion object {
        /**
         * 参数ids ',' 分隔转list<Integer>
         *
         * @param ids string ids
         * @return list<Integer>
        <*/
        @JvmStatic
        fun StringIdsToList(ids: String): List<Int> {
            val idArr = ids.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return idArr.map { NumberUtils.toInt(it) }
        }

        /**
         * 参数ids ',' 分隔转list<String>
         *
         * @param ids string ids
         * @return list<String>
         */
        @JvmStatic
        fun StringIdsToStringList(ids: String): List<String> {
            val idArr = ids.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val newIds = ArrayList<String>()
            Collections.addAll(newIds, *idArr)
            return newIds
        }

        /**
         * 参数ids ',' 是否为Integer类型
         *
         * @param ids string ids
         * @return true or false
         */
        @JvmStatic
        fun StringIdsIsNumber(ids: String): Boolean {
            val idArr = ids.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return idArr.any { NumberUtils.isNumber(it) }
        }
    }
}